### ISSUE TITLE
[Spellcheck] Not taking .dic and .aff files

### ISSUE THREAD

**Issue by jytou** [geany/geany-plugins#552]
I have downloaded dictionaries for French (typically from [here](https://extensions.libreoffice.org/extensions/dictionnaires-francais) and [there](http://extensions.openoffice.org/en/project/french-dictionaries)), unzipped in a directory where I can see that the subdirectory "dictionaries" contain .aff and .dic files. But then when I point the spellchecker plugin to that "dictionaries" directory, still even after restarting geany, no new language is appearing in the spellchecking menus.

**Comment by eht16** [comment]
I'm afraid those dictionary files are broken or in some way not supported by Enchant/MySpell.
Just tried the Bulgarian dictionary files from http://extensions.services.openoffice.org/en/dictionaries (chose Bulgarian just for fun, it was the first link on the site I found) and the included .aff and .dic files worked fine.

I tried the version 5.7 of the French dictionaries and they didn't work. I cannot say why they didn't work, unfortunately Enchant does not tell why it won't load the files. Though I straced the Geany process and could see that Enchant *does* try to load those files.
After I renamed `fr-moderne.aff` and `fr-moderne.dic` to `fr_FR.aff` (and .dic accordingly), they got loaded. Don't ask me why it seems necessary to name them as the country code but this seems to do the trick.
This is nothing the SpellCheck plugin can change.


Background: Enchant is a spell checking library which is used by the SpellCheck plugin.
Enchant itself supports various backends which are responsible for providing the dictionary data. The only such backend which supports loading dictionaries from a specified path is the MySpell backend.
